src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/SummaryReportService.java

```diff
@@ -62,6 +68,28 @@ public void process(LocalDate date) {
 
         EnvelopeSupplierStatement envelopeSupplierStatement = optSupplierStatement.get();
 
+        var existingReports = reconciliationReportRepository
+            .findByStatementId(envelopeSupplierStatement.id);
+
+        List<ReconciliationReport> existingReportsDistinctByAccount =
+            existingReports
+                .stream()
+                .filter(distinctByKey(r -> r.account))
+                .collect(Collectors.toList());
+
+        if (existingReportsDistinctByAccount.size() != existingReports.size()) {
```

Comment:
can we not just do 
`if(existingReports.size() != TargetStorageAccount.values().length)`?

Reply:
if there is more than one report for a account `existingReports` would be wrong to compare. Normally it should not but added to stay in secure side.